scripts/gen_sdcard_image.sh

```diff
@@ -19,16 +13,21 @@ BOOTFSSIZE="$4"
 ROOTFSSIZE="$5"
 UBOOT="$6"
 
+[ -z "$SOURCE_DATE_EPOCH" ] || \
+	SIGNATURE="-S $(printf "%.8d" $SOURCE_DATE_EPOCH)"
```

Comment:
Missing `0x` before signature string, without that `ptgen` assigns random value.
Also this should be handled literally as string, because with current setting it doesn't truncate the provided `$SOURCE_DATE_EPOCH` and `ptgen` takes last eight digits. This breaks booting of ESPRESSObin which relies on the same string but is handled like this:
https://github.com/openwrt/openwrt/blob/dc629d9cf5afc03f67a294078ca1001e831c66c9/target/linux/mvebu/image/Makefile#L16
This takes first eight digits from `$SOURCE_DATE_EPOCH`.

Reply:
hmm ; i'll use that;
i tried to run the command separately and it gave me different results ; but i'll update it if you say it works with `$(shell printf "%.8s" $(SOURCE_DATE_EPOCH)) `

Reply:
You still forgot about `0x`, the line should look like this:
`SIGNATURE="-S 0x$(printf "%.8d" $SOURCE_DATE_EPOCH)"`

Reply:
my bad; i misunderstood ; will update

Reply:
Sorry, You didn't misunderstand, I copy-pasted the old line and added `0x`, the correct one shoul look like this:
`SIGNATURE="-S 0x$(printf "%.8s" $SOURCE_DATE_EPOCH)"`
Sorry again for adding to the confusion.

Reply:
no worries;
thanks for the update :)

updated PR